src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs

```diff
@@ -91,33 +91,28 @@ internal OtlpLogs.LogRecord ToOtlpLog(LogRecord logRecord)
             var attributeValueLengthLimit = this.sdkLimitOptions.LogRecordAttributeValueLengthLimit;
             var attributeCountLimit = this.sdkLimitOptions.LogRecordAttributeCountLimit ?? int.MaxValue;
 
-            /*
-            // Removing this temporarily for stable release
-            // https://github.com/open-telemetry/opentelemetry-dotnet/issues/4776
-            // https://github.com/open-telemetry/opentelemetry-dotnet/issues/3491
-            // First add the generic attributes like Category, EventId and Exception,
-            // so they are less likely being dropped because of AttributeCountLimit.
-
-            if (!string.IsNullOrEmpty(logRecord.CategoryName))
+            if (this.experimentalOptions.EmitLogEventAndCategoryAttributes)
             {
-                // TODO:
-                // 1. Track the following issue, and map CategoryName to Name
-                // if it makes it to log data model.
-                // https://github.com/open-telemetry/opentelemetry-specification/issues/2398
-                // 2. Confirm if this name for attribute is good.
-                otlpLogRecord.AddStringAttribute("dotnet.ilogger.category", logRecord.CategoryName, attributeValueLengthLimit, attributeCountLimit);
-            }
+                if (!string.IsNullOrEmpty(logRecord.CategoryName))
+                {
+                    // TODO:
+                    // 1. Track the following issue, and map CategoryName to Name
+                    // if it makes it to log data model.
+                    // https://github.com/open-telemetry/opentelemetry-specification/issues/2398
+                    // 2. Confirm if this name for attribute is good.
+                    AddStringAttribute(otlpLogRecord, "dotnet.ilogger.category", logRecord.CategoryName, attributeValueLengthLimit, attributeCountLimit);
+                }
 
-            if (logRecord.EventId.Id != default)
-            {
-                otlpLogRecord.AddIntAttribute(nameof(logRecord.EventId.Id), logRecord.EventId.Id, attributeCountLimit);
-            }
+                if (logRecord.EventId.Id != default)
+                {
+                    AddIntAttribute(otlpLogRecord, "event.id", logRecord.EventId.Id, attributeCountLimit);
```

Comment:
@ThomsonTan Could you update how OTel Cpp exports its eventid, given OTel Cpp is the only other language with eventid being a first class thing?

Reply:
EventId is currently ignored when exporting to OTLP in OTel Cpp. I thought we should create a semantic convention for it to make the implementation across languages consistent? And don't add any language specific name to the attribute name.

Reply:
Yes that makes sense. Given this is experimental, it'd be okay to change the attribute name from event.id to something else (if the spec/sem. convention warrants it).

@vishweshbankwar Could you open a spec issue to create a semantic convention for event.id (a numerical identifier to be attached to individual logs), and add a comment linking to that?